Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XRSDK/Controllers/WindowsMixedRealityXRSDKArticulatedHand.cs

```diff
@@ -0,0 +1,325 @@
+using UnityEngine.XR.WindowsMR;
+using Windows.Perception.People;
+using Windows.UI.Input.Spatial;
+#endif // WINDOWS_UWP
+
+namespace Microsoft.MixedReality.Toolkit.XRSDK.WindowsMixedReality
+{
+    /// <summary>
+    /// XR SDK implementation of Windows Mixed Reality articulated hands.
+    /// </summary>
+    [MixedRealityController(
+        SupportedControllerType.ArticulatedHand,
+        new[] { Handedness.Left, Handedness.Right })]
+    public class WindowsMixedRealityXRSDKArticulatedHand : BaseWindowsMixedRealityXRSDKSource, IMixedRealityHand
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        public WindowsMixedRealityXRSDKArticulatedHand(TrackingState trackingState, Handedness controllerHandedness, IMixedRealityInputSource inputSource = null, MixedRealityInteractionMapping[] interactions = null)
+            : base(trackingState, controllerHandedness, inputSource, interactions)
+        {
+        }
+
+        /// <inheritdoc />
+        public override MixedRealityInteractionMapping[] DefaultInteractions => new[]
+        {
+            new MixedRealityInteractionMapping(0, "Spatial Pointer", AxisType.SixDof, DeviceInputType.SpatialPointer),
+            new MixedRealityInteractionMapping(1, "Spatial Grip", AxisType.SixDof, DeviceInputType.SpatialGrip),
+            new MixedRealityInteractionMapping(2, "Select", AxisType.Digital, DeviceInputType.Select),
+            new MixedRealityInteractionMapping(3, "Grab", AxisType.SingleAxis, DeviceInputType.TriggerPress),
+            new MixedRealityInteractionMapping(4, "Index Finger Pose", AxisType.SixDof, DeviceInputType.IndexFinger)
+        };
+
+        private static readonly HandFinger[] handFingers = Enum.GetValues(typeof(HandFinger)) as HandFinger[];
+        private readonly List<Bone> fingerBones = new List<Bone>();
+        private readonly Dictionary<TrackedHandJoint, MixedRealityPose> unityJointPoses = new Dictionary<TrackedHandJoint, MixedRealityPose>();
+
+#if WINDOWS_UWP
+        private HandMeshObserver handMeshObserver = null;
+        private int[] handMeshTriangleIndices = null;
+        private bool hasRequestedHandMeshObserver = false;
+        private Vector2[] handMeshUVs;
+
+        protected void InitializeUVs(Vector3[] neutralPoseVertices)
+        {
+            if (neutralPoseVertices.Length == 0)
+            {
+                Debug.LogError("Loaded 0 verts for neutralPoseVertices");
+            }
+
+            float minY = neutralPoseVertices[0].y;
+            float maxY = minY;
+
+            for (int ix = 1; ix < neutralPoseVertices.Length; ix++)
+            {
+                Vector3 p = neutralPoseVertices[ix];
+
+                if (p.y < minY)
+                {
+                    minY = p.y;
+                }
+                else if (p.y > maxY)
+                {
+                    maxY = p.y;
+                }
+                float d = p.x * p.x + p.y * p.y;
```

Comment:
Is d used anywhere after this computation?

Reply:
It sure doesn't look like it! This was copied over from https://github.com/microsoft/MixedRealityToolkit-Unity/blame/mrtk_development/Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XR2018/WindowsMixedRealityArticulatedHand.cs#L187.

It's used in the linked class but...not really. I'll clean this up in both places (and continue investigating how we can better base class these, e.g. https://github.com/microsoft/MixedRealityToolkit-Unity/pull/6973#issuecomment-577360562)